File: Lexor/Lexor.Model/Responses/PayrollSettingsResponse.cs
namespace Lexor.Model.Responses
{
    public class PayrollSettingsResponse
    {
        public int Id { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime? ValidTo { get; set; }
        public string WorkDaysDescription { get; set; } = string.Empty;
        public int WorkDaysMask { get; set; }
        public decimal OvertimeMultiplier { get; set; }
        public decimal PersonalDeduction { get; set; }
        public decimal PioMioRate { get; set; }
        public decimal HealthInsuranceRate { get; set; }
        public decimal UnemploymentRate { get; set; }
        public decimal IncomeTaxRate { get; set; }
    }
}

File: Lexor/UI/lexor_desktop/lib/models/payroll_settings_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'payroll_settings_response.g.dart';

@JsonSerializable(createToJson: false)
class PayrollSettingsResponse {
  int id;
  DateTime validFrom;
  DateTime? validTo;
  String workDaysDescription;
  int workDaysMask;
  double overtimeMultiplier;
  double personalDeduction;
  double pioMioRate;
  double healthInsuranceRate;
  double unemploymentRate;
  double incomeTaxRate;

  PayrollSettingsResponse({
    required this.id,
    required this.validFrom,
    this.validTo,
    required this.workDaysDescription,
    required this.workDaysMask,
    required this.overtimeMultiplier,
    required this.personalDeduction,
    required this.pioMioRate,
    required this.healthInsuranceRate,
    required this.unemploymentRate,
    required this.incomeTaxRate,
  });

  factory PayrollSettingsResponse.fromJson(Map<String, dynamic> json) =>
      _$PayrollSettingsResponseFromJson(json);
}
